File: C/3142-check-if-grid-satisfies-conditions/main.go
func satisfiesConditions(grid [][]int) bool {
	for i := 0; i < len(grid); i++ {
		for j := 0; j < len(grid[0]); j++ {
			if i < len(grid)-1 && grid[i][j] != grid[i+1][j] || j < len(grid[0])-1 && grid[i][j] == grid[i][j+1] {
				return false
			}
		}
	}
	return true
}

File: C/3142-check-if-grid-satisfies-conditions/main.rs
struct Solution {}

impl Solution {
  pub fn satisfies_conditions(grid: Vec<Vec<i32>>) -> bool {
    let mut row: usize = 0;
    while row < grid.len() {
      let mut col: usize = 0;
      while col < grid[0].len() {
        if row < grid.len() - 1 && grid[row][col] != grid[row + 1][col]
          || col < grid[0].len() - 1 && grid[row][col] == grid[row][col + 1]
        {
          return false;
        }
        col += 1;
      }
      row += 1;
    }
    true
  }
}
